interpretation/src/main/kotlin/com/hartwig/actin/treatment/EvidenceScoringModel.kt

```diff
@@ -21,14 +21,25 @@ data class EvidenceScore(
 
 class EvidenceScoringModel(val config: ScoringConfig) {
 
-    fun score(treatment: TreatmentEvidence): EvidenceScore {
+    fun score(
+        treatment: TreatmentEvidence,
+        isIndirect: Boolean = false
+    ): EvidenceScore {
         val cancerTypeApplicability = when (treatment.cancerTypeMatch.applicability) {
             CancerTypeMatchApplicability.SPECIFIC_TYPE -> TumorMatch.PATIENT
             CancerTypeMatchApplicability.ALL_TYPES -> TumorMatch.ALL
             CancerTypeMatchApplicability.OTHER_TYPE -> TumorMatch.ANY
         }
-        val exactVariant = if (treatment.molecularMatch.sourceEvidenceType.isCategoryEvent()) VariantMatch.CATEGORY else VariantMatch.EXACT
-        val scoringMatch = ScoringMatch(cancerTypeApplicability, exactVariant)
+        val isCategoryEvent = treatment.molecularMatch.sourceEvidenceType.isCategoryEvent()
+        check(!(isIndirect && isCategoryEvent)) {
```

Comment:
Could this ever happen in production without a programming error? What should someone do if this error occurs?